doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst

```diff
@@ -0,0 +1,61 @@
+==================
+A four bar linkage
+==================
+
+The four bar linkage is a common example used in mechanics, which has to holonomic constraints. This example will make use of joints functionality provided in :mod:`sympy.physics.mechanics`. In summary we will use bodies and joints to define the open loop system. Next, we define the configuration constraints to close the loop. With the ``JointsMethod`` we get the input to manually setup the ``KanesMethod`` as its backend.
```

Comment:
A good habit to start is to make these references: `:py:class:~JointsMethod` or something like that. I think there are examples in the documentation style guide.

Reply:
The `JointsMethod` is not added to the docstring yet. Currently I'm thinking about making a new PR in which:
- [The Joint (Docstrings) page](https://docs.sympy.org/dev/modules/physics/mechanics/api/joint.html) is renamed to `Joints Framework (Docstrings)` (better name for framework is welcome).
   - Here I would also add the `JointsMethod` docstring
- Add a separate guide to the [guide to mechanics](https://docs.sympy.org/dev/modules/physics/mechanics/index.html#guide-to-mechanics) called `Joints framework in Physics/Mechanics`. This guide will include:
   - An explanation of the purposes of the different classes used (e.g. `Joint`, `PinJoint`, `JointsMethod`)
   - The explanation mentioned in #21705

I would put the images about each of the joints (like shown [here](https://tjstienstra.github.io/gsoc/2022/2022/05/21/my-first-post.html) for the `CylindricalJoint`)) in the docstrings.

@sidhu1012 and @moorepants what do you think of this proposal?

Reply:
Created a [draft PR](https://github.com/sympy/sympy/pull/23730) for it.